src/Language/PureScript/Docs/Convert/Single.hs

```diff
@@ -106,6 +108,26 @@ augmentDeclarations (partitionEithers -> (augments, toplevels)) =
       mergeComments kc Nothing = kc
       mergeComments (Just kcoms) (Just dcoms) =
         Just $ kcoms <> "\n" <> dcoms
+  augmentWith (AugmentRole comms roles) d =
+    d { declComments = mergeComments (declComments d) comms
+      , declInfo = insertRoles
+      }
+    where
+      mergeComments Nothing rc = rc
+      mergeComments dc Nothing = dc
+      mergeComments (Just dcoms) (Just rcoms) =
+        Just $ dcoms <> "\n" <> rcoms
+
+      insertRoles = case declInfo d of
+        DataDeclaration dataDeclType args [] ->
+          DataDeclaration dataDeclType args roles
+        TypeSynonymDeclaration args ty [] ->
+          TypeSynonymDeclaration args ty roles
+        DataDeclaration _ _ _ ->
+          P.internalError "augmentWith: could not add a second role to data declaration"
+        TypeSynonymDeclaration _ _ _ ->
```

Comment:
One question, can FFI have role declarations? I think it can, but I couldn't remember. I'm pretty sure it defaults to nominal unless otherwise stated, correct? If it can, I need to add a case for `ExternDeclaration` here.

Reply:
Yes https://github.com/purescript/purescript-effect/blob/985d97bd5721ddcc41304c55a7ca2bb0c0bfdc2a/src/Effect.purs#L18

Reply:
Thanks. I've added them.